perf-tool/src/monitor.cpp

```diff
@@ -81,25 +76,25 @@ JNIEXPORT void JNICALL MonitorContendedEntered(jvmtiEnv *jvmtiEnv, JNIEnv *env,
     j["Class"] = str;
     
     /* free str */
-
-
-    numContentions[str].numFirstTier++;
-
-    int num = numContentions[str].numFirstTier;
-    j["numTypeContentions"] = num;
+   
+    static std::mutex contentionMutex;
+    int samplesPerObjectType = 0;
+    {
+    std::lock_guard<std::mutex> lg(contentionMutex);
+    samplesPerObjectType = ++(numContentions[std::string(str)]);
```

Comment:
Is this never decremented anywhere? Also, is the value by default 0 the first time the key `std::string(str)` is used?

Reply:
> Is this never decremented anywhere?

Correct. The value is ever increasing.

> Also, is the value by default 0 the first time the key std::string(str) is used?

Yes. When the key is not found, a new {key, value} is inserted and for `value` the default constructor is used, which in this case will set the `int` to 0.